icelake/src/types/in_memory.rs

```diff
@@ -2007,6 +2004,186 @@ impl Snapshot {
+    const SPARK_APP_ID: &'static str = "spark.app.id";
+    const ADDED_DATA_FILES: &'static str = "added-data-files";
+    const TOTAL_DATA_FILES: &'static str = "total-data-files";
+    const ADDED_DELETE_FILES: &'static str = "added-delete-files";
+    const ADDED_EQUALITY_DELETE_FILES: &'static str = "added-equality-delete-files";
+    const ADDED_POSITION_DELETE_FILES: &'static str = "added-position-delete-files";
+    const TOTAL_DELETE_FILES: &'static str = "total-delete-files";
+    const ADDED_RECORDS: &'static str = "added-records";
+    const TOTAL_RECORDS: &'static str = "total-records";
+    const ADDED_POSITION_DELETES: &'static str = "added-position-deletes";
+    const TOTAL_POSITION_DELETES: &'static str = "total-position-deletes";
+    const ADDED_EQUALITY_DELETES: &'static str = "added-equality-deletes";
+    const TOTAL_EQUALITY_DELETES: &'static str = "total-equality-deletes";
+    const ADDED_FILES_SIZE: &'static str = "added-files-size";
+    const TOTAL_FILES_SIZE: &'static str = "total-files-size";
+
+    pub fn new() -> Self {
+        Default::default()
+    }
+
+    pub fn update(&mut self, datafile: &DataFile) {
+        match datafile.content {
+            DataContentType::Data => {
+                self.added_data_files += 1;
+                self.added_data_records += datafile.record_count;
+                self.added_files_size += datafile.file_size_in_bytes;
+            }
+            DataContentType::PostionDeletes => {
+                self.added_delete_files += 1;
+                self.added_position_delete_files += 1;
+                self.added_position_deletes_records += datafile.record_count;
+                self.added_files_size += datafile.file_size_in_bytes;
+            }
+            DataContentType::EqualityDeletes => {
+                self.added_delete_files += 1;
+                self.added_equality_delete_files += 1;
+                self.added_equality_deletes_records += datafile.record_count;
+                self.added_files_size += datafile.file_size_in_bytes;
+            }
+        }
+    }
+
+    fn operation(&self) -> String {
+        if self.added_delete_files == 0 && self.added_data_files != 0 {
+            "append".to_string()
+        } else if self.added_delete_files != 0 && self.added_data_files != 0 {
+            "overwrite".to_string()
+        } else if self.added_delete_files != 0 && self.added_data_files == 0 {
+            "delete".to_string()
+        } else {
+            "append".to_string()
+        }
+    }
+
+    pub fn merge(
+        self,
+        last_summary: &HashMap<String, String>,
+        is_compact_op: bool,
+    ) -> Result<HashMap<String, String>> {
+        let operation = if is_compact_op {
+            "replace".to_string()
+        } else {
+            self.operation()
+        };
+
+        #[inline]
+        fn get_i64(value: &HashMap<String, String>, key: &str) -> std::result::Result<i64, Error> {
```

Comment:
I think we should not return error, but a default value here?

Reply:
Yes. The error is for parse::<i64>